ethereum/core/src/main/java/tech/pegasys/teku/core/StateTransition.java

```diff
@@ -77,52 +76,52 @@ public BeaconState initiate(
         preState,
         signed_block,
         validateStateRootAndSignatures,
-        interimState -> {},
         IndexedAttestationProvider.DIRECT_PROVIDER);
   }
 
   public BeaconState initiate(
       BeaconState preState,
-      SignedBeaconBlock signed_block,
+      SignedBeaconBlock signedBlock,
       boolean validateStateRootAndSignatures,
-      final Consumer<BeaconState> beaconStateConsumer,
       final IndexedAttestationProvider indexedAttestationProvider)
       throws StateTransitionException {
     try {
-      BlockValidator blockValidator =
-          validateStateRootAndSignatures ? this.blockValidator : BlockValidator.NOOP;
-      final BeaconBlock block = signed_block.getMessage();
-
       // * Process slots (including those with no blocks) since block
       // * beaconStateConsumer only consumes the missing slots here,
       //   the new block will be processed when adding to the store.
-      BeaconState postSlotState =
-          process_slots(
-              preState,
-              block.getSlot(),
-              state -> {
-                if (!state.getSlot().equals(block.getSlot())) {
-                  beaconStateConsumer.accept(state);
-                }
-              });
+      BeaconState postSlotState = process_slots(preState, signedBlock.getMessage().getSlot());
+
+      return processAndValidateBlock(
+          signedBlock, indexedAttestationProvider, postSlotState, validateStateRootAndSignatures);
+    } catch (SlotProcessingException | EpochProcessingException | IllegalArgumentException e) {
+      LOG.warn("State Transition error", e);
```

Comment:
dont want this at error level? seems bad...

Reply:
It was always warn level.  I'm sceptical of logging and error and throwing it but figured trying to change that was outside the scope of this PR.